data/src/test/java/io/gnosis/data/repositories/TransactionRepositoryTest.kt

```diff
@@ -0,0 +1,391 @@
+package io.gnosis.data.repositories
+
+import io.gnosis.data.backend.TransactionServiceApi
+import io.gnosis.data.models.*
+import io.gnosis.data.repositories.TokenRepository.Companion.ETH_SERVICE_TOKEN_INFO
+import io.gnosis.data.repositories.TokenRepository.Companion.FAKE_ERC20_TOKEN_INFO
+import io.gnosis.data.repositories.TokenRepository.Companion.FAKE_ERC721_TOKEN_INFO
+import io.mockk.coEvery
+import io.mockk.coVerify
+import io.mockk.mockk
+import kotlinx.coroutines.test.runBlockingTest
+import org.junit.Assert.*
+import org.junit.Test
+import pm.gnosis.crypto.utils.asEthereumAddressChecksumString
+import pm.gnosis.model.Solidity
+import pm.gnosis.utils.asDecimalString
+import pm.gnosis.utils.asEthereumAddress
+import java.math.BigInteger
+
+class TransactionRepositoryTest {
+
+    private val transactionServiceApi = mockk<TransactionServiceApi>()
+
+    private val transactionRepository = TransactionRepository(transactionServiceApi)
+    private val defaultSafeAddress = "0x1C8b9B78e3085866521FE206fa4c1a67F49f153A".asEthereumAddress()!!
+    private val defaultFromAddress = "0x7cd310A8AeBf268bF78ea16C601F201ca81e84Cc".asEthereumAddress()!!
+    private val defaultToAddress = "0x2134Bb3DE97813678daC21575E7A77a95079FC51".asEthereumAddress()!!
+    private val defaultValue = BigInteger("230000000000000000")
+
+    @Test
+    fun `getTransactions (api failure) should throw`() = runBlockingTest {
+        val throwable = Throwable()
+        coEvery { transactionServiceApi.loadTransactions(any()) } throws throwable
+
+        val actual = runCatching { transactionRepository.getTransactions(defaultSafeAddress) }
+
+        with(actual) {
+            assert(isFailure)
+            assertEquals(throwable, exceptionOrNull())
+        }
+        coVerify(exactly = 1) { transactionServiceApi.loadTransactions(defaultSafeAddress.asEthereumAddressChecksumString()) }
+    }
+
+    @Test
+    fun `getTransactions (module transaction) should return custom`() = runBlockingTest {
+        val transactionDto = buildModuleTransactionDto(module = Solidity.Address(BigInteger.ONE))
+        val pagedResult = listOf(transactionDto)
+        coEvery { transactionServiceApi.loadTransactions(any()) } returns Page(1, null, null, pagedResult)
+
+        val actual = transactionRepository.getTransactions(defaultSafeAddress)
+
+        with(actual.results[0] as Transaction.Custom) {
+            assertEquals(transactionDto.module, address)
+            assertEquals(transactionDto.data?.dataSizeBytes() ?: 0L, dataSize)
```

Comment:
Why is it necessary to have a default value for these checks? I think that if something is null, it should be null also in the other side of the mapping.

Reply:
Or hardcode the default value in the repository so we get more clarity what the default value is in the tests.

Reply:
Custom.dataSize() currently can't be null. Because it is of type Long. Also it is computed and not in the dto. I would prefer *not* to make dataSize nullable. 

The default is necessary because data can be null. But IMHO dataSize should be 0 then, and not null.

Reply:
yeah but for the assertion I mean. You shouldn't have `?:` in the values you are asserting.